### PR TITLE
Hotfixz 114 missing company id after migration

### LINKED ISSUE #114: Missing company id after migration

### PR DESCRIPTION
**Pull request by anttileppa**
Resolves #114

### CODE REVIEW

File: src/holdings/holdings.py [hunk 1]
```diff
@@ -38,6 +39,31 @@ def add_holding(self, security_id: UUID, holding_date: date, amount: Decimal):
...
+
+            if start_date > security_min_date:
+                start_date = security_min_date
+
+            total = Decimal(0)
+            for i in range((end_date - start_date).days + 1):
+                holding_date = start_date + timedelta(days=i)
+
+                change = self.data[security_id].get(holding_date, None)
+                if change is not None:
```
Comment (baherami) [inline src/holdings/holdings.py:62]: fix prev pr comments if possible.

**Review by baherami — APPROVED** [review]

File: src/alembic/versions/0017_company_updated.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# revision identifiers, used by Alembic.
+revision = '0017'
+down_revision = '0016'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column("company", sa.Column("updated", sa.DateTime, nullable=False))
```
Comment (baherami) [inline src/alembic/versions/0017_company_updated.py:19]: add to the models too please. Add default value too :)

**Review by baherami — CHANGES_REQUESTED** [review]

File: src/commands/migration_tasks.py [hunk 3]
```diff
@@ -604,43 +604,47 @@ class MigrateCompaniesTask(AbstractFundsTask):
...
+            self.source_updated = self.get_source_last_update_date(funds_session=funds_session)
+            self.backend_updated = self.get_backend_last_update_date(backend_session=backend_session)
 
-        Returns: backend database company count
+            if self.source_updated is None:
+                self.source_updated = datetime(1970, 1, 1, 0, 0)
 
-        """
-        return backend_session.query(func.count(destination_models.Company.id)).scalar()
+            if self.backend_updated is None:
```
Comment (baherami) [inline src/commands/migration_tasks.py:631]: cant be null but it can be unix time as default.

File: src/commands/migration_tasks.py [hunk 4]
```diff
@@ -604,43 +604,47 @@ class MigrateCompaniesTask(AbstractFundsTask):
     Migration task for companies
     """
 
+    def __init__(self):
+        """
+        Constructor
+        """
+        self.source_updated = None
+        self.backend_updated = None
```
Comment (baherami) [inline src/commands/migration_tasks.py:612]: is always unix time as default( if there is a default value for it based on alembic migration).

Comment (anttileppa) [reply src/commands/migration_tasks.py:612]: True, but the value needs to be defined because it's loaded in prepare method, not constructor

**Review by baherami — APPROVED** [review]

**Pull request merged**